src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+
+        public LibgpiodCapiDriver()
+        {
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        public LibgpiodCapiDriver(string chip)
+        {
+            if (chip != null)
+                GpioChip = chip;
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        private void OpenChip()
+        {
+            _chip = Interop.gpiod_chip_open($"{BasePath}{GpioChip}");
+            if (_chip == null)
+            {
+                throw new GpioChipException($"Chip: {GpioChip} not available, error code: {Marshal.GetLastWin32Error()}");
+            }
+        }
+
+        private void RequestEvent(int pinNumber, LibgpiodDriverEventHandler eventDescriptor, PinEventTypes eventType)
+        {
+            int eventSuccess = -1;
+            SafeLineHandle pin = Interop.gpiod_chip_get_line(_chip, pinNumber);
+            if (pin != null)
+            {
+                eventDescriptor.PinHandle = pin;
+                if (eventType == PinEventTypes.Rising)
+                {
+                    eventSuccess = Interop.gpiod_line_request_rising_edge_events(pin, $"Listen {pinNumber} for rising edge event");
+                }
+                else if (eventType == PinEventTypes.Falling)
+                {
+                    eventSuccess = Interop.gpiod_line_request_falling_edge_events(pin, $"Listen {pinNumber} for falling edge event");
+                }
+            }
+            if (eventSuccess < 0)
+            {
+                throw new GpioChipException($"Error while requesting event listener for pin {pinNumber}, error code: {Marshal.GetLastWin32Error()}");
+            }
+        }
+
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            if (!_pinNumberToEventHandler.TryGetValue(pinNumber, out LibgpiodDriverEventHandler eventHandler))
+            {
+                eventHandler = new LibgpiodDriverEventHandler(pinNumber, new CancellationTokenSource());
+                _pinNumberToEventHandler.Add(pinNumber, eventHandler);
+                RequestEvent(pinNumber, eventHandler, eventType);
+                InitializeEventDetectionThread(eventHandler);
+            }
+            if (eventType.HasFlag(PinEventTypes.Rising))
+                eventHandler.ValueRising += callback;
+            if (eventType.HasFlag(PinEventTypes.Falling))
+                eventHandler.ValueFalling += callback;
+        }
+
+        private void InitializeEventDetectionThread(LibgpiodDriverEventHandler eventHandler)
+        {
+            Thread pinListeningThread = new Thread(DetectLineEvent)
+            {
+                IsBackground = true
+            };
+
+            pinListeningThread.Start(eventHandler);
+        }
+
+        private void DetectLineEvent(Object obj)
```

Comment:
why is the arg not directly LibgpiodDriverEventHandler?

Reply:
I don't really know the exact reason, when having arg other than object type Thread constractor not accepting that method name DetectLineEvent (`new Thread(DetectLineEvent)`)

Reply:
that issue might go away if you decide to convert to task